### ISSUE TITLE
Steam apps can't select proper resolution

### ISSUE THREAD

**Issue by dinotheextinct** [labwc/labwc#1790]
Hello,
I recently upgraded to a 4k monitor and I am not able to chose the correct resolution in Steamgame. Regardless which game I choose it offers me 2560x1440 OR 2841x1598 as highest option (which is weird resolution and might give insight on the problem?) 

I think it might have to do something with HiDPI scaling? I uncommented all changes to labwc config for the scaling, but games still don't offer me the proper 4k resolution.

labwc version 0.7.1
Opensuse Tumbleweed

**Comment by Consolatis** [comment]
When `wlr-randr` shows a scale factor of `1`, does the game then provide the correct resolution?

**Comment by dinotheextinct** [comment]
Yes it does. So how can I achieve both?

**Comment by ahesford** [comment]
You ask the author of the game for proper scaling support, or you manually switch to unit scale before you start the game. `wdisplays` is a helpful tool. I don't think this is a labwc problem.

**Comment by dinotheextinct** [comment]
Its not a single game, its all games with steam.

**Comment by ahesford** [comment]
Then that sounds like a problem with Steam.

**Comment by dinotheextinct** [comment]
Just fyi. the problem doesn't exist on KDE Wayland session.

**Comment by droc12345** [comment]
KDE is not a wlroots based compositor and they might very well have added workarounds for steam.

Have you tried any other wlroots based compositor, sway, hyprland, wayfire, etc?

**Comment by ahesford** [comment]
It's come to my attention that Steam is an X11 application, and the problem you're seeing is common to all X11 applications with Xwayland.

**Comment by dinotheextinct** [comment]
Yes exactly and it makes labwc unuseable as desktop WM if you have a HiDPI screen. If I understand the settings correctly you are describing in the manual, the wlr-randr scale should not affect XWayland applications. For that purpose you have the xprop and GDK Scale options? Here is how KDE is doing it: https://wiki.archlinux.org/title/HiDPI#Xwayland_2

Also the scaling for apps like signal-desktop from flatpak looks much more crisp on KDE with scaling like this:
![image](https://github.com/labwc/labwc/assets/32158581/824298b1-38ff-4680-ab0b-53f351ef8052)

**Comment by ahesford** [comment]
If it's unusable, use something else. This is not a labwc issue.

**Comment by Consolatis** [comment]
While technically correct, this is kind of a wlroots and thus labwc issue.

**Comment by winteriscariot** [comment]
For games, using [gamescope](https://software.opensuse.org/package/gamescope?locale=en) is an option as it has built-in scaling options independent from labwc

**Comment by dinotheextinct** [comment]
> For games, using [gamescope](https://software.opensuse.org/package/gamescope?locale=en) should be a viable workaround as it has built-in scaling options independent from labwc

Sadly I haven't found a solution with gamescope which works for me. With capital width and height the game still doesn't offer the correct resolution in game or doesn't even start properly (the later is probably a problem of the game).

**Comment by sascha-wi** [comment]
Might this solve this issue in the future? https://www.phoronix.com/news/XWayland-Rootful-HiDPI-Merged

Also is there any work on the integration of this or how does one check if rootful xwayland is supported by labwc?

**Comment by Consolatis** [comment]
rootful xwayland is supported by labwc (and should be the same for any other wayland compositor as its basically just a regular wayland client):

```
Xwayland -decorate -noreset :55
DISPLAY=:55 dbus-run-session openbox-session
```

Rather than starting `openbox-session` you could obviously start whatever you like.
Note that this will not show the windows within that X11 session in `A-Tab`, render labwc decorations or similar. It might be fine to run that fullscreen for a single game or similar though, never tested that approach and also can't say anything about the performance.

**Comment by dinotheextinct** [comment]
I am not even able to start any game with gamescope anymore. Works without problems on KDE:

**Comment by DreamMaoMao** [comment]
https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/2064
https://gitlab.freedesktop.org/xorg/xserver/-/issues/1318

This is a problem that has always existed with wlroots
